### ISSUE TITLE
First breadcrumb is always collapsed into overflow ellipsis

### ISSUE THREAD

**Issue by micahgodbolt** [OfficeDev/office-ui-fabric-react#1277]
via Kyle Curlett

In the outlook web app we are using breadcrumbs in our file picker. We are encountering an issue where the **first breadcrumb is always collapsed into overflow ellipsis** (and actual overflow isn't). I think the problem lies in BreadCrumb.updateRenderedItems, specifically the parts in bold below. 

    Breadcrumb.prototype._updateRenderedItems = function () {
        var _a = this.props, items = _a.items, maxDisplayedItems = _a.maxDisplayedItems;
        **var renderingArea = this.refs.renderingArea;**
        var renderedItems = [];
        var renderedOverflowItems = [].concat(items);
        var consumedWidth = 0;
        var style = window.getComputedStyle(renderingArea);
        **var availableWidth = renderingArea.clientWidth - parseInt(style.marginLeft, 10) - parseInt(style.marginRight, 10);**
        **availableWidth -= OVERFLOW_WIDTH;**
        var i;
        var minIndex = Math.max(0, renderedOverflowItems.length - maxDisplayedItems);
        for (i = renderedOverflowItems.length - 1; i >= minIndex; i--) {
            var item = renderedOverflowItems[i];
            var itemWidth = this._breadcrumbItemWidths[item.key];
            **if ((consumedWidth + itemWidth) >= availableWidth) {
                break;
            }**
            else {
                consumedWidth += itemWidth;
            }
        }
        renderedItems = renderedOverflowItems.splice(i + 1);
        this.setState({
            isOverflowOpen: this.state.isOverflowOpen,
            overflowAnchor: this.state.overflowAnchor,
            renderedItems: renderedItems,
            renderedOverflowItems: renderedOverflowItems,
        });
    };

At least in our case, the renderingArea is being set to the area needed to render the objects. Then we are subtracting off the overflow width, and saying that if we go above this (smaller) width we will start collapsing breadcrumbs. This leads to the first 54 pixels always being collapsed, and nothing else. Is this the intended behavior? If not, **what parameter do we need to be passing in/using to set the correct renderingArea so that the overflow will work correctly?** this.refs.renderingArea is set in BreadCrumb.render and it looks to me like it will always be the space needed to render the breadcrumb, but it seems to be used as if it were the space available to render breadcrumbs.

**Comment by micahgodbolt** [comment]
First pass. The logic appears to be working on our website example. If you remote the maxDisplayedItems you get

![image](https://cloud.githubusercontent.com/assets/1434956/23956757/bdc8213e-095a-11e7-9e39-8f8e2a53cf7f.png)


I can also confirm the proper behavior via a codepen http://codepen.io/micahgodbolt/pen/aJLZrm

**Comment by micahgodbolt** [comment]
If I'm reading this correctly, "the renderingArea is being set to the area needed to render the objects", so you are measuring the space needed for the breadcrumb, and then setting the renderingArea to that size?

So what you might be looking for is a way to turn off the breadcrumb collapsing?

**Comment by kcurlett** [comment]
Hello Micah, thanks for the reply.

Clarification: for the part in quotes, I meant that the renderingArea is automatically being set to the sum of the breadcrumb widths, instead of the size of the container they live in. 

We do want the breadcrumbs to collapse, but something is happening that is giving us the incorrect rendering area, so they are collapsing incorrectly. 

For an example lets say that the container the breadcrumbs live in is 500px, and the breadcrumbs are 100px each. We would want them to collapse if there were 6 (or even 5) breadcrumbs. But in the above code I noticed that rederingArea.clientWidth is being set to (100px * numberOfBreadCrumbs), regardless of container size, which is causing the problem. 

As far as I can tell, we aren't setting or modifying renderingArea or renderingArea.clientWidth in our code, they are being automatically generated. I am not sure how to change these so that they will use the width of the container, which is what they should be, and what they seem to be elsewhere. 

The incorrect renderingArea is clearly related to something we are doing, since your code seems to work elsewhere. I'm not sure what it is we are doing incorrectly, or how to fix it. 

This issue has been moved to low priority on my team, so if finding a solution would require a lot of digging you do not need to worry about it. 

Thank you again,
Kyle
